src/commands/createFunction.ts

```diff
@@ -47,20 +58,165 @@ export async function createFunction(outputChannel: vscode.OutputChannel): Promi
+            }
+        }
+    } else {
+        switch (setting.valueType) {
+            case ValueType.boolean:
+                return await promptForBooleanSetting(ui, setting);
+            default:
+        }
+    }
+
+    // Default to 'string' type for any setting that isn't supported
+    return await promptForStringSetting(ui, setting, defaultValue);
+}
+
+async function promptForSubscription(ui: IUserInterface, azureAccount: AzureAccount): Promise<AzureResourceFilter | undefined> {
+    if (azureAccount.filters.length === 0) {
+        return undefined;
+    } else if (azureAccount.filters.length === 1) {
+        return azureAccount.filters[0];
+    } else {
+        const subscriptionPicks: PickWithData<AzureResourceFilter>[] = [];
+        azureAccount.filters.forEach((f: AzureResourceFilter) => {
+            if (f.subscription.displayName) {
+                subscriptionPicks.push(new PickWithData<AzureResourceFilter>(f, f.subscription.displayName));
+            }
+        });
+        const placeHolder: string = localize('azFunc.selectSubscription', 'Select a Subscription');
+
+        return (await ui.showQuickPick<AzureResourceFilter>(subscriptionPicks, placeHolder)).data;
+    }
+}
+
+async function promptForDocumentDB(ui: IUserInterface, functionAppPath: string, credentials: ServiceClientCredentials, subscriptionId: string, setting: ConfigSetting): Promise<string> {
+    const client: CosmosDBManagementClient = new CosmosDBManagementClient(credentials, subscriptionId);
+    const dbAccount: DatabaseAccount = await showResourceQuickPick<DatabaseAccount>(ui, setting, client.databaseAccounts.list());
+    if (dbAccount.id && dbAccount.name) {
+        const resourceGroup: string = azUtil.getResourceGroupFromId(dbAccount.id);
+        const keys: DatabaseAccountListKeysResult = await client.databaseAccounts.listKeys(resourceGroup, dbAccount.name);
+        const appSettingName: string = `${dbAccount.name}_${ResourceType.DocumentDB.toUpperCase()}`;
+        await setLocalFunctionAppSetting(functionAppPath, appSettingName, `AccountEndpoint=${dbAccount.documentEndpoint};AccountKey=${keys.primaryMasterKey};`);
+
+        return appSettingName;
+    } else {
+        throw new Error(localize('azFunc.InvalidCosmosDBAccount', 'Invalid Cosmos DB Account'));
+    }
+}
+
+async function promptForBooleanSetting(ui: IUserInterface, setting: ConfigSetting): Promise<string> {
+    const picks: Pick[] = [new Pick('true'), new Pick('false')];
+
+    return (await ui.showQuickPick(picks, setting.label, false)).label;
+}
+
+async function promptForStringSetting(ui: IUserInterface, setting: ConfigSetting, defaultValue?: string): Promise<string> {
+    const prompt: string = localize('azFunc.stringSettingPrompt', 'Provide a {0}', setting.label);
+    defaultValue = defaultValue ? defaultValue : setting.defaultValue;
+
+    return await ui.showInputBox(setting.label, prompt, false, (s: string) => setting.validateSetting(s), defaultValue);
```

Comment:
I tried it, but I get a type error because 'this' is undefined inside validateSetting

---
In reply to: [147786296](https://github.com/Microsoft/vscode-azurefunctions/pull/32#discussion_r147786296) [](ancestors = 147786296)